scrapers/nv/events.py

```diff
@@ -1,82 +1,167 @@
+    r"(SJR|AR|AJR|IP|SCR|SB|ACR|SR|AB)\s{0,5}0*(\d+)", flags=re.IGNORECASE
+)
 
-    # Note: The NV Site has three different calendars.
-    # https://www.leg.state.nv.us/MeetingDisplay/CalendarOfMeetings/
-    # is the primary calendar. The title is 2011-2012 but shows current events.
-    # It appears to have the same information and more compared to
-    # https://www.leg.state.nv.us/App/Calendar/A/
-    # The the social calendar is at
-    # https://www.leg.state.nv.us/socialCalendar/socialCalendarDisplay.cfm?date=05/01/2017
-    # and is not being scraped
+date_re = re.compile(r"([a-z]+ \d{1,2}, .* (am|pm))", flags=re.IGNORECASE)
 
-    def scrape(self):
-        html = self.get(self.URL).text
-        page = lxml.html.fromstring(html)
-
-        for row in page.xpath('//table[@width="98%" and @cellpadding="3"]/tr'):
-            yield self.scrape_event(row)
-
-    def scrape_event(self, row):
-        date_td = row.xpath("td[1]")[0]
-        info_td = row.xpath("td[2]")[0]
-
-        date = date_td.xpath("b")[0].text.strip()
-        time = date_td.xpath("b/following-sibling::text()")[0].strip()
-
-        date_and_time = "{} {}".format(date, time)
-        start_date = datetime.datetime.strptime(date_and_time, "%m/%d/%y %I:%M %p")
-
-        title = info_td.xpath("font[1]/strong")[0].text.strip()
-
-        all_text = info_td.xpath("descendant-or-self::*/text()")
-        notes = (line.strip() for line in all_text if line.strip())
-        notes = list(notes)
-
-        if len(notes) > 1:
-            # Skip the first line, which is the title
-            notes = notes[1:]
-            # Split out the address
-            address = notes[0]
-            notes = notes[1:]
-            # The rest just becomes the description
-            notes = "\n".join(notes)
-        else:
-            address = "TBD"
-            notes = notes[0]
-
-        event = Event(
-            start_date=self._TZ.localize(start_date),
-            name=title,
-            location_name=address,
-            description=notes,
+start_time_re = re.compile(r"Start\s*Time\s*(\d+)\s*(AM|PM)", flags=re.IGNORECASE)
+
+
+class AgendaStartTime(PdfPage):
+    def process_page(self):
+        # Simplify text to make regex simpler
+        self.text = self.text.replace(". ", "").replace(".", "")
+
+        start = start_time_re.search(self.text)
+        if not start:
+            # Default to 12am if no start time is found
+            yield "12:00 am"
+            return
+        time, ampm = start.groups(1)
+        yield f"{time}:00 {ampm}"
+
+
+class Agenda(PdfPage):
+    def process_page(self):
+        # Simplify text to make regex simpler
+        self.text = self.text.replace(". ", "").replace(".", "")
+
+        # Find all bill ids
+        bills = bills_re.findall(self.text)
+
+        # Format bills before yielding
+        formatted_bills = set()
+        for alpha, num in bills:
+            formatted_bills.add(f"{alpha.upper()} {num}")
```

Comment:
Clever way to dodge duplicates or just a more complicated way to say:

```
        for alpha, num in bills:
            yield f"{alpha.upper()} {num}")
```
?

Reply:
My intent was to prevent duplicates

Reply:
Cool. Just making sure. Thanks!